src/NSrtm.Core/Pgm/DataDesciption/PgmDataDescriptionExtractor.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace NSrtm.Core.Pgm
+{
+    internal static class PgmPreamble
+    {
+        public static readonly IReadOnlyDictionary<string, Regex> Extractor =
+            new Dictionary<string, Regex>
+            {
+                {"MagicNumber", new Regex(@"^(P5)[^$]", RegexOptions.Multiline)},
+                {"Offset", new Regex(@"^#[\s]Offset[\s]+(?<offset>.+)$", RegexOptions.Multiline)},
+                {"Scale", new Regex(@"^#[\s]Scale[\s]+(?<scale>.+)$", RegexOptions.Multiline)},
+                {"Origin", new Regex(@"^#[\s]Origin[\s](?<lat>\d+)\D+(?<lon>\d+)", RegexOptions.Multiline)},
```

Comment:
Are you sure?

This preamble : P5
# Geoid file in PGM format for the GeographicLib::Geoid class
# Description WGS84 EGM84, 30-minute grid
# URL http://earth-info.nga.mil/GandG/wgs84/gravitymod/wgs84_180/wgs84_180.html
# DateTime 2009-08-29 18:45:02
# RMSBilinearError 0.070
# Offset -108
# MaxBilinearError 1.546
# MaxCubicError 0.274
# RMSCubicError 0.014
# Scale 0.003
# Origin 90 0
# AREA_OR_POINT Point
# Vertical_Datum WGS84
720  361
65535

passed regex tester http://regexstorm.net/tester and my UT. Can you be more specific and give me your preamble version?